src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -82,25 +82,22 @@ export class LocalKernelFinder implements ILocalKernelFinder {
     @captureTelemetry(Telemetry.KernelFinderPerf)
     public async findKernel(
         resource: Resource,
-        option?: nbformat.INotebookMetadata,
+        notebookMetadata?: nbformat.INotebookMetadata,
         cancelToken?: CancellationToken
     ): Promise<LocalKernelConnectionMetadata | undefined> {
         try {
             // Get list of all of the specs
             const kernels = await this.listKernels(resource, cancelToken);
             const isPythonNbOrInteractiveWindow =
-                isPythonNotebook(option) || getResourceType(resource) === 'interactive';
+                isPythonNotebook(notebookMetadata) || getResourceType(resource) === 'interactive';
 
             // Always include the interpreter in the search if we can
             const preferredInterpreter =
-                option && isInterpreter(option)
-                    ? option
-                    : resource && isPythonNbOrInteractiveWindow && this.extensionChecker.isPythonExtensionInstalled
+                resource && isPythonNbOrInteractiveWindow && this.extensionChecker.isPythonExtensionInstalled
```

Comment:
the type of `option` was at some point changed, hence some of the code is no longer valid.
Took me a while to figure that out (hence removed old code, not valid)